### PR TITLE
Implement pango markup for -window-format

### PR DESCRIPTION
**Pull request by jsmnbom**
This PR implements pango markup for -window-format as requested in issue #1287 (see that issue for how to use).

As discussed in IRC, i believe I've done stuff mostly correct, but there might very well be some memory leaks and other issues, since I've never really worked with C and most of what I've done is guesswork...

### CODE REVIEW

File: source/dialogs/window.c [hunk 1]
```diff
@@ -337,11 +337,14 @@ static client* window_client ( ModeModePrivateData *pd, xcb_window_t win )
...
     }
-    c->title      = g_markup_escape_text( c->title, -1 );
+    c->title      = g_markup_escape_text ( c->title, -1 );
     pd->title_len = MAX ( c->title ? g_utf8_strlen ( c->title, -1 ) : 0, pd->title_len );
 
-    c->role      = window_get_text_prop ( c->window, netatoms[WM_WINDOW_ROLE] );
-    c->role     = g_markup_escape_text( c->role, -1 );
+    c->role = window_get_text_prop ( c->window, netatoms[WM_WINDOW_ROLE] );
+    if ( c->role == NULL ) {
+        c->role = "";
```
Comment (DaveDavenport) [inline source/dialogs/window.c:345]: This will cause problems when `g_free` ed later.

File: source/dialogs/window.c [hunk 2]
```diff
@@ -337,11 +337,14 @@ static client* window_client ( ModeModePrivateData *pd, xcb_window_t win )
...
+    }
+    c->role      = g_markup_escape_text ( c->role, -1 );
```
Comment (DaveDavenport) [inline source/dialogs/window.c:347]: leaks the result of value in c->role that gets overwritten.

File: source/dialogs/window.c [hunk 3]
```diff
@@ -337,11 +337,14 @@ static client* window_client ( ModeModePrivateData *pd, xcb_window_t win )
     if ( c->title == NULL ) {
         c->title = window_get_text_prop ( c->window, XCB_ATOM_WM_NAME );
     }
-    c->title      = g_markup_escape_text( c->title, -1 );
+    c->title      = g_markup_escape_text ( c->title, -1 );
```
Comment (DaveDavenport) [inline source/dialogs/window.c:340]: leaks the result of value in c->title  that gets overwritten.

**Comment by jsmnbom** [comment]
Alright, I believe I've fixed those issues now. Is it also an issue with https://github.com/davatorium/rofi/pull/1288/files#diff-27b5f68bbbead2a38c679b28a0c8ba5e25ec8e81abd02fcd4f77012833f07f77R353 and https://github.com/davatorium/rofi/pull/1288/files#diff-27b5f68bbbead2a38c679b28a0c8ba5e25ec8e81abd02fcd4f77012833f07f77R355 or are those fine?

File: source/dialogs/window.c [hunk 4]
```diff
@@ -332,20 +333,26 @@ static client* window_client ( ModeModePrivateData *pd, xcb_window_t win )
...
     }
 
-    c->title = window_get_text_prop ( c->window, xcb->ewmh._NET_WM_NAME );
-    if ( c->title == NULL ) {
-        c->title = window_get_text_prop ( c->window, XCB_ATOM_WM_NAME );
+    char *tmp_title = window_get_text_prop ( c->window, xcb->ewmh._NET_WM_NAME );
+    if ( tmp_title == NULL ) {
+        tmp_title = window_get_text_prop ( c->window, XCB_ATOM_WM_NAME );
     }
+    c->title      = g_markup_escape_text ( tmp_title, -1 );
```
Comment (jsmnbom) [inline source/dialogs/window.c:340]: Hmm.. is it possible to tmp_title to still be NULL here? Should it maybe be `tmp_title ? tmp_title : ""`?

**Comment by DaveDavenport** [comment]
FYI, been busy, will get back to reviewing this soon.

**Comment by jsmnbom** [comment]
Oh yea, that's totally okay, i understand :3

File: source/dialogs/window.c [hunk 5]
```diff
@@ -332,20 +333,26 @@ static client* window_client ( ModeModePrivateData *pd, xcb_window_t win )
...
+    c->role        = g_markup_escape_text ( tmp_role ? tmp_role : "", -1 );
+    pd->role_len   = MAX ( c->role ? g_utf8_strlen ( c->role, -1 ) : 0, pd->role_len );
+    g_free ( tmp_role );
 
     cky = xcb_icccm_get_wm_class ( xcb->connection, c->window );
     xcb_icccm_get_wm_class_reply_t wcr;
     if ( xcb_icccm_get_wm_class_reply ( xcb->connection, cky, &wcr, NULL ) ) {
         c->class     = rofi_latin_to_utf8_strdup ( wcr.class_name, -1 );
+        c->class     = g_markup_escape_text( c->class, -1 );
         c->name      = rofi_latin_to_utf8_strdup ( wcr.instance_name, -1 );
```
Comment (DaveDavenport) [inline source/dialogs/window.c:354]: This leaks memory. (c->class). confirmed with valgrind.

File: source/dialogs/window.c [hunk 6]
```diff
@@ -332,20 +333,26 @@ static client* window_client ( ModeModePrivateData *pd, xcb_window_t win )
...
+        c->name      = g_markup_escape_text( c->name, -1 );
```
Comment (DaveDavenport) [inline source/dialogs/window.c:355]: same here.

**Comment by jsmnbom** [comment]
Oh wow I had no idea valgrind was a thing, that could've come in handy earlier 😅.
I believe there's no more memory leaks now.

**Comment by DaveDavenport** [comment]
thanks.

**Pull request merged**